test/hotspot/jtreg/compiler/c2/TestOptimizeSubstring.java

```diff
@@ -70,8 +70,9 @@ public static void main (String args[]) {
 
     private static void check(boolean enabled) {
         OutputAnalyzer oa;
-        String newStringAlloc = "call,static  wrapper for: _new_array_nozero_Java";
-
+        // ideally, we should check the opcode 'call' as well, but x86_32.ad uses all CAPITAL opcodes.
+        // the feature is 'CALL, static  wrapper for: _new_array_nozero_Java' on i686.
+        String newStringAlloc = /*call ,*/"static  wrapper for: _new_array_nozero_Java";
         try {
             oa = ProcessTools.executeTestJvm("-XX:+UnlockDiagnosticVMOptions", "-Xbootclasspath/a:.",
```

Comment:
Is this comment identifying a future TODO item?  That's not entirely clear to me.

Reply:
It's not a TODO.  I try to explain why I remove the opcode call here. it's because x86_32.ad uses the capital opcode CALL, which is different from all other architectures.   Unfortunately, `oa.shouldNotContain` and `oa.shouldContain` don't support case-insensitive comparison. 

Without the opcode,  newStringAlloc is still a unique and portable pattern to check `OptimizeSubstring` takes effect or not.